### PR TITLE
unitem:::fit.ctxs doesn't exist, but fit$ctxs does

### PR DESCRIPTION
**Pull request by nlhepler**

### CODE REVIEW

**Comment by nlhepler** [comment]
@dalexander this works, and is the last thing ayang needs to run trainings again

**Comment by nlhepler** [comment]
here's some ipython evidence for the latter commit:

```
In [1]: import re

In [2]: fname = "xsf0.0.chunk0.subreadset.xml"

In [3]: fields = fname.split(".")

In [4]: if re.match(r"chunk[0-9]+"):
   ...:     print("".join(fields[:-2]))
   ...: else:
   ...:     print(".".join(fields[:-1]))
   ...:     
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-eaf9f69f18a4> in <module>()
----> 1 if re.match(r"chunk[0-9]+"):
      2     print("".join(fields[:-2]))
      3 else:
      4     print(".".join(fields[:-1]))
      5 

TypeError: match() takes at least 2 arguments (1 given)

In [5]: if re.match(r"chunk[0-9]+", fields[-2]):
   ...:     print(".".join(fields[:-2]))
   ...: else:
   ...:     print(".".join(fields[:-1]))
   ...:     
xsf0.0.chunk0.subreadset

In [6]: fields[-2]
Out[6]: 'subreadset'

In [7]: if re.match(r"chunk[0-9]+", fields[-3]):
   ...:     print(".".join(fields[:-3]))
   ...: else:
   ...:     print(".".join(fields[:-2]))
   ...:     
xsf0.0

In [8]: fields = "xsf0.0.subreadset.xml".split(".")

In [9]: if re.match(r"chunk[0-9]+", fields[-3]):
   ...:     print(".".join(fields[:-3]))
   ...: else:
   ...:     print(".".join(fields[:-2]))
   ...:     
xsf0.0

In [10]: if re.match(r"chunk[0-9]+", fields[-3]):
    ...:     print(".".join(fields[:-3]))
    ...: else:
    ...:     print(".".join(fields[:-2]))
    ...:     
xsf0.0

In [11]: fields = "xsf0.0.chunk100.subreadset.xml".split(".")

In [12]: if re.match(r"chunk[0-9]+", fields[-3]):
    ...:     print(".".join(fields[:-3]))
    ...: else:
    ...:     print(".".join(fields[:-2]))
    ...:     
xsf0.0

In [13]: fields = "movieName.chunk100.subreadset.xml".split(".")

In [14]: if re.match(r"chunk[0-9]+", fields[-3]):
    ...:     print(".".join(fields[:-3]))
    ...: else:
    ...:     print(".".join(fields[:-2]))
    ...:     
movieName
```

File: bauhaus/workflows/secondary/datasetOps.py [hunk 1]
```diff
@@ -13,8 +14,11 @@ def movieName(filename):
 
     base = op.basename(filename)
     fields = base.split(".")
-    assert len(fields) in (3, 4) and fields[-1] == "xml"
-    movieName = fields[0]
+    assert len(fields) > 2 and fields[-1] == "xml"
+    if len(fields) > 3 and re.match(r"chunk[0-9]+", fields[-3]):
+        movieName = ".".join(fields[:-3])
+    else:
+        movieName = ".".join(fields[:-2])
```
Comment (dalexander) [inline bauhaus/workflows/secondary/datasetOps.py:21]: With this change the method comment is no longer quite accurate.  The comment could be amended by removing/adjusting the first sentence; the _example_ is still accurate.

On balance, I'm not sure about merging this change.  Technically, PacBio movie names do not contain "." characters, and I could argue that setting up filenames in this manner is a violation of bauhaus's implicit assumptions.  The solution, instead, would be to make such implicit assumptions _explicit_, rejecting such data at the `validate` step.

Comment (dalexander) [reply bauhaus/workflows/secondary/datasetOps.py:21]: I'm going to merge for the moment this to unblock Alicia but I may come back to this later with some different thoughts, and I'll adjust the comment.

**Pull request merged**